File: external/HDF-EOS/hdfeos/samples/UpdateLevels.c
#include "mfhdf.h"

/*   UpdateLevels */

main()
{

    intn            status, i, j;
    int32           ptfid;
    int32           PTid1, PTid2;
    int32           nrec, recs[32];
    int32           outNrec, outRecs[32];
    int32           inLevel, outLevel;
    
    char            datbuf[1024];
    float64         f64;
    

    /*
     * Open the HDF point file, "PointFile.hdf".
     */

    ptfid = PTopen("PointFile.hdf", DFACC_RDWR);


    if (ptfid != -1)
    {

	PTid1 = PTattach(ptfid, "Simple Point");
	PTid2 = PTattach(ptfid, "FixedBuoy Point");

	if (PTid1 != -1 && PTid2 != -1)
	{

	    f64 = 43.2;
	    memcpy(datbuf, &f64, 8);
	    recs[0] = 1;
	    
	    status = PTupdatelevel(PTid2, 0, "Longitude", 1, recs, datbuf);


	    datbuf[0] = 'F';
	    nrec = 1;
	    recs[0] = 0;
	    status = PTupdatelevel(PTid2, 0, "ID", nrec, recs, datbuf);

	    inLevel = 0;
	    outLevel = 1;
	    status = PTgetrecnums(PTid2, inLevel, outLevel, nrec, recs, 
				  &outNrec, outRecs);


	    for (i=0; i<outNrec; i++)
	    {
		datbuf[i] = 'F';
	    }

	    status = PTupdatelevel(PTid2, outLevel, "ID", 
				   outNrec, outRecs, datbuf);

	}
    }

    PTdetach(PTid1);
    PTdetach(PTid2);

    PTclose(ptfid);
    HEprint(stdout,0);
    
    return;
}

File: external/HDF-EOS/hdfeos/samples/updatelevels.f
	program updatelevels

	integer            ptuplev, ptdetach, ptclose, ptgetrecnums
	integer*4          ptfid, ptid1, ptid2
	integer*4          recs(32), inlevel, outlevel
	integer*4          ptopen, ptattach
	integer*4          outrecs(32), outnrec
	real*8             f64
	
	character          datbuf*256, c8*8
	equivalence        (f64,c8)
	
	integer DFACC_RDWR
	parameter (DFACC_RDWR=3)

c
c     Open the HDF point file, "PointFile.hdf".
c

	ptfid = ptopen("PointFile.hdf", DFACC_RDWR)

	ptid1 = ptattach(ptfid, "Simple Point")
	ptid2 = ptattach(ptfid, "FixedBuoy Point")

	f64 = 43.2
	datbuf(1:8) = c8
	recs(1) = 1
	
	status = ptuplev(ptid2, 0, "Longitude", 1, recs, datbuf)


	datbuf(1:1) = 'F'
	recs(1) = 0
	status = ptuplev(ptid2, 0, "ID", 1, recs, datbuf)
	    
	inlevel = 0
	outlevel = 1
	nrec = 1
	status = ptgetrecnums(ptid2, inlevel, outlevel, nrec, recs, 
     1			      outnrec, outrecs)

	
	do 10 i=1,outnrec
		datbuf(i:i) = 'F'
 10	continue

	status = ptuplev(ptid2, outlevel, "ID", outnrec, outrecs, datbuf)

	status = ptdetach(ptid1)
	status = ptdetach(ptid2)
	
	status = ptclose(ptfid)
	
	stop
	end

